charts/tf-controller/templates/deployment.yaml

```diff
@@ -46,12 +49,12 @@ spec:
             fieldRef:
               fieldPath: metadata.namespace
         - name: RUNNER_POD_IMAGE
-          value: "{{ .Values.runner.image.repository }}:{{ .Values.runner.image.tag}}"
+          value: "{{ .Values.runner.image.repository }}:{{ default $.Chart.AppVersion .Values.runner.image.tag }}"
```

Comment:
I don't think you need `$` as it is not in a particular scope and should be able to access `.Chart` without it.